### PR TITLE
fixing several tests failing in mocha locally

### PR DESCRIPTION
**Pull request by bfitzpat**
Signed-off-by: bfitzpat@redhat.com <bfitzpat@redhat.com>

### CODE REVIEW

**Comment by bfitzpat** [comment]
@lhein Can you give this a once-over and I'll get it merged, then rebase my other PR to make sure I didn't break any other tests in it

**Comment by bfitzpat** [comment]
I have four tests failing locally, with what appear to be sinon/spy-related problems. I cannot duplicate the issues on travis, nor have I found a way to work around them locally. But they are consistent. Open to any suggestions if you have any.  Thanks

```
 78 passing (54s)

  4 failing

  1) Kubectl integration watcher
       Check there is one message logged in case of connection error:
     Error: Timeout of 20000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/bfitzpat/github/vscode-camelk/out/src/test/suite copy/kubectlwatcher.test.js)
  	at listOnTimeout (internal/timers.js:531:17)
  	at processTimers (internal/timers.js:475:7)


  2) Kubectl integration watcher
       Check there is only one message logged in case of connection error with View visible:
     Error: Timeout of 20000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/bfitzpat/github/vscode-camelk/out/src/test/suite copy/kubectlwatcher.test.js)
  	at listOnTimeout (internal/timers.js:531:17)
  	at processTimers (internal/timers.js:475:7)


  3) ensure install methods are functioning as expected
       install Camel K and Kubernetes CLIs on activation:
     TypeError: Attempted to wrap installKubectl which is already wrapped
  	at checkWrappedMethod (node_modules/sinon/lib/sinon/util/core/wrap-method.js:40:21)
  	at wrapMethod (node_modules/sinon/lib/sinon/util/core/wrap-method.js:88:13)
  	at Function.spy (node_modules/sinon/lib/sinon/spy.js:157:16)
  	at Sandbox.spy (node_modules/sinon/lib/sinon/sandbox.js:322:35)
  	at Context.<anonymous> (out/src/test/suite/install.test.js:39:43)
  	at Generator.next (<anonymous>)
  	at /home/bfitzpat/github/vscode-camelk/out/src/test/suite/install.test.js:24:71
  	at new <anonymous> (/home/bfitzpat/.vscode/extensions/vscjava.vscode-java-test-0.22.2/dist/extension.bundle.js:1:275159)
  	at __awaiter (out/src/test/suite/install.test.js:20:12)
  	at Context.<anonymous> (out/src/test/suite/install.test.js:38:16)
  	at processImmediate (internal/timers.js:439:21)
  --------------
  Error: Stack Trace for original
  	at wrapMethod (node_modules/sinon/lib/sinon/util/core/wrap-method.js:116:26)
  	at Function.spy (node_modules/sinon/lib/sinon/spy.js:157:16)
  	at Sandbox.spy (node_modules/sinon/lib/sinon/sandbox.js:322:35)
  	at Suite.<anonymous> (out/src/test/suite copy/install.test.js:36:35)
  	at Object.<anonymous> (out/src/test/suite copy/install.test.js:35:1)
  	at Module.o._compile (.vscode-test/vscode-1.42.1/VSCode-linux-x64/resources/app/out/vs/loader.js:15:749)
  	at Object.Module._extensions..js (internal/modules/cjs/loader.js:798:10)
  	at Object.Module._extensions.<computed> [as .js] (node_modules/istanbul/lib/hook.js:109:37)
  	at Module.load (internal/modules/cjs/loader.js:645:32)
  	at Function.Module._load (internal/modules/cjs/loader.js:560:12)
  	at Function.t._load (.vscode-test/vscode-1.42.1/VSCode-linux-x64/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:837:268)
  	at Function.n._load (.vscode-test/vscode-1.42.1/VSCode-linux-x64/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:807:106)
  	at Function.i._load (.vscode-test/vscode-1.42.1/VSCode-linux-x64/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:803:550)
  	at Module.require (internal/modules/cjs/loader.js:685:19)
  	at n (.vscode-test/vscode-1.42.1/VSCode-linux-x64/resources/app/out/vs/loader.js:15:524)
  	at Array.forEach (<anonymous>)
  	at /home/bfitzpat/github/vscode-camelk/out/src/test/suite/index.js:58:23
  	at f (node_modules/once/once.js:25:25)
  	at Glob.<anonymous> (node_modules/glob/glob.js:151:7)
  	at Glob._finish (node_modules/glob/glob.js:197:8)
  	at done (node_modules/glob/glob.js:182:14)
  	at Glob._processGlobStar2 (node_modules/glob/glob.js:637:12)
  	at /home/bfitzpat/github/vscode-camelk/node_modules/glob/glob.js:626:10
  	at RES (node_modules/inflight/inflight.js:31:16)
  	at f (node_modules/once/once.js:25:25)
  	at lstatcb_ (node_modules/glob/glob.js:519:7)
  	at RES (node_modules/inflight/inflight.js:31:16)
  	at f (node_modules/once/once.js:25:25)
  	at FSReqCallback.oncomplete (fs.js:166:5)


  4) Kubectl integration watcher
       "before all" hook for "Check there is no loop for closing kubectl process":
     TypeError: Attempted to wrap shareMessage which is already wrapped
  	at checkWrappedMethod (node_modules/sinon/lib/sinon/util/core/wrap-method.js:40:21)
  	at wrapMethod (node_modules/sinon/lib/sinon/util/core/wrap-method.js:88:13)
  	at Function.spy (node_modules/sinon/lib/sinon/spy.js:157:16)
  	at Sandbox.spy (node_modules/sinon/lib/sinon/sandbox.js:322:35)
  	at Context.<anonymous> (out/src/test/suite/kubectlwatcher.test.js:46:30)
  	at processImmediate (internal/timers.js:439:21)
  --------------
  Error: Stack Trace for original
  	at wrapMethod (node_modules/sinon/lib/sinon/util/core/wrap-method.js:116:26)
  	at Function.spy (node_modules/sinon/lib/sinon/spy.js:157:16)
  	at Sandbox.spy (node_modules/sinon/lib/sinon/sandbox.js:322:35)
  	at Suite.<anonymous> (out/src/test/suite copy/kubectlwatcher.test.js:35:28)
  	at Object.<anonymous> (out/src/test/suite copy/kubectlwatcher.test.js:34:1)
  	at Module.o._compile (.vscode-test/vscode-1.42.1/VSCode-linux-x64/resources/app/out/vs/loader.js:15:749)
  	at Object.Module._extensions..js (internal/modules/cjs/loader.js:798:10)
  	at Object.Module._extensions.<computed> [as .js] (node_modules/istanbul/lib/hook.js:109:37)
  	at Module.load (internal/modules/cjs/loader.js:645:32)
  	at Function.Module._load (internal/modules/cjs/loader.js:560:12)
  	at Function.t._load (.vscode-test/vscode-1.42.1/VSCode-linux-x64/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:837:268)
  	at Function.n._load (.vscode-test/vscode-1.42.1/VSCode-linux-x64/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:807:106)
  	at Function.i._load (.vscode-test/vscode-1.42.1/VSCode-linux-x64/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:803:550)
  	at Module.require (internal/modules/cjs/loader.js:685:19)
  	at n (.vscode-test/vscode-1.42.1/VSCode-linux-x64/resources/app/out/vs/loader.js:15:524)
  	at Array.forEach (<anonymous>)
  	at /home/bfitzpat/github/vscode-camelk/out/src/test/suite/index.js:58:23
  	at f (node_modules/once/once.js:25:25)
  	at Glob.<anonymous> (node_modules/glob/glob.js:151:7)
  	at Glob._finish (node_modules/glob/glob.js:197:8)
  	at done (node_modules/glob/glob.js:182:14)
  	at Glob._processGlobStar2 (node_modules/glob/glob.js:637:12)
  	at /home/bfitzpat/github/vscode-camelk/node_modules/glob/glob.js:626:10
  	at RES (node_modules/inflight/inflight.js:31:16)
  	at f (node_modules/once/once.js:25:25)
  	at lstatcb_ (node_modules/glob/glob.js:519:7)
  	at RES (node_modules/inflight/inflight.js:31:16)
  	at f (node_modules/once/once.js:25:25)
  	at FSReqCallback.oncomplete (fs.js:166:5)
```

**Comment by bfitzpat** [comment]
I am a little concerned that I get this locally...
```
  Suite duration: 0.001 s, Tests: 1
  78 passing (56s)
  4 failing
```

and this on Travis -- where are the other 25 tests?
```
  Suite duration: 0.009 s, Tests: 1
  43 passing (8s)
```

**Comment by bfitzpat** [comment]
This also brings in a few dependabot changes to fix:

* https://github.com/camel-tooling/vscode-camelk/pull/336 
* https://github.com/camel-tooling/vscode-camelk/pull/334
* https://github.com/camel-tooling/vscode-camelk/pull/333
* https://github.com/camel-tooling/vscode-camelk/pull/332
* https://github.com/camel-tooling/vscode-camelk/pull/331

File: src/test/suite/completion.java.test.ts [hunk 1]
```diff
@@ -19,22 +19,23 @@
...
-
 	const expectedCompletion = { label: 'from(String uri) : RouteDefinition'};
 
-	var testVar = test('Completes from method for Java', async () => {
-		if(os.homedir().includes('hudson')) {
-			testVar.skip();
-		}
-		await testCompletion(docUriJava, new vscode.Position(5, 11), expectedCompletion);
-	}).timeout(76000);
+	test('java completion method', function() {
```
Comment (apupier) [inline src/test/suite/completion.java.test.ts]: writing it this way is no more checking anything. The test is always passing in 0ms, it doesn't wait for the check to happen

**Comment by bfitzpat** [comment]
@apupier Though I would love to point to specific tests failing, without the changes I made in the PR, I can't get ANY tests to run locally. That was the initial problem with trying to diagnose the test in https://github.com/camel-tooling/vscode-camelk/pull/330. I can't debug the tests if I can't run them.

On master I get...

```
$ npm run test

> vscode-camelk@0.0.13 test /home/bfitzpat/github/vscode-camelk
> npm run compile && node ./out/src/test/runTests.js


> vscode-camelk@0.0.13 compile /home/bfitzpat/github/vscode-camelk
> tsc -p ./ && npm run copy-sanitize


> vscode-camelk@0.0.13 copy-sanitize /home/bfitzpat/github/vscode-camelk
> copyfiles -u 1 src/test/suite/sanitize.go.saved out/src

extensionDevelopmentPath = /home/bfitzpat/github/vscode-camelk
extensionTestsPath = /home/bfitzpat/github/vscode-camelk/out/src/test/suite/index
Downloading VS Code 1.43.0 from https://update.code.visualstudio.com/1.43.0/linux-x64/stable
Downloaded VS Code 1.43.0 into .vscode-test/vscode-1.43.0
vscodeExecutablePath = /home/bfitzpat/github/vscode-camelk/.vscode-test/vscode-1.43.0/VSCode-linux-x64/code
Installing extensions...
Extension 'ms-kubernetes-tools.vscode-kubernetes-tools' is already installed.
VS Code extension ms-kubernetes-tools.vscode-kubernetes-tools installed
Installing extensions...
Extension 'redhat.java' is already installed.
VS Code extension redhat.java installed
Gtk-Message: 06:35:49.549: Failed to load module "appmenu-gtk-module"

Warning: 'sandbox' is not in the list of known options, but still passed to Electron/Chromium.

[main 2020-03-11T12:35:49.923Z] update#setState idle

bash: cannot set terminal process group (-1): Inappropriate ioctl for device
bash: no job control in this shell

(node:16300) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information

(node:16300) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information

(node:16300) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information

(node:16300) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information

(node:16300) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information

(node:16300) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information

Didact configuration cleared
/home/bfitzpat/.vscode/extensions/redhat.vscode-apache-camel-0.0.22/jars/language-server.jar

testsRoot = /home/bfitzpat/github/vscode-camelk/out/src/test

TypeError: Attempted to wrap installKubectl which is already wrapped

TypeError: Attempted to wrap installKubectl which is already wrapped

(node:16452) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information

(node:16452) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information

You registered cluster type aks

You registered cluster type acs

You registered cluster type minikube

Exit code:   1
Done

Failed to run testsFailed
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! vscode-camelk@0.0.13 test: `npm run compile && node ./out/src/test/runTests.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the vscode-camelk@0.0.13 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/bfitzpat/.npm/_logs/2020-03-11T12_35_56_622Z-debug.log

```

The log reports:
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'test' ]
2 info using npm@6.14.2
3 info using node@v12.16.1
4 verbose run-script [ 'pretest', 'test', 'posttest' ]
5 info lifecycle vscode-camelk@0.0.13~pretest: vscode-camelk@0.0.13
6 info lifecycle vscode-camelk@0.0.13~test: vscode-camelk@0.0.13
7 verbose lifecycle vscode-camelk@0.0.13~test: unsafe-perm in lifecycle true
8 verbose lifecycle vscode-camelk@0.0.13~test: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/bfitzpat/github/vscode-camelk/node_modules/.bin:/home/bfitzpat/.config/Code/User/globalStorage/redhat.vscode-camelk/camelk/tools/kamel/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
9 verbose lifecycle vscode-camelk@0.0.13~test: CWD: /home/bfitzpat/github/vscode-camelk
10 silly lifecycle vscode-camelk@0.0.13~test: Args: [ '-c', 'npm run compile && node ./out/src/test/runTests.js' ]
11 silly lifecycle vscode-camelk@0.0.13~test: Returned: code: 1  signal: null
12 info lifecycle vscode-camelk@0.0.13~test: Failed to exec test script
13 verbose stack Error: vscode-camelk@0.0.13 test: `npm run compile && node ./out/src/test/runTests.js`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:311:20)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:311:20)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
14 verbose pkgid vscode-camelk@0.0.13
15 verbose cwd /home/bfitzpat/github/vscode-camelk
16 verbose Linux 5.3.0-7629-generic
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "test"
18 verbose node v12.16.1
19 verbose npm  v6.14.2
20 error code ELIFECYCLE
21 error errno 1
22 error vscode-camelk@0.0.13 test: `npm run compile && node ./out/src/test/runTests.js`
22 error Exit status 1
23 error Failed at the vscode-camelk@0.0.13 test script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

**Comment by apupier** [comment]
I don't understand the link between the errors that you have on master and the modified code in this PR.

Do you have failures when launching the tests from VS Code directly? or only using command-line `npm run test`?

I tried with the same combination of npm and node and it is "working" (I have a test failing locally but it is starting the test execution).

Can you try to modify the index.ts file and add more traces. it seems that the last log available is coming from there, just before loadCoverageRunner is called.
Then can you add more traces in loadCoverageRunner and try to remove the coverage folder before launching the test.

File: src/test/suite/completion.java.test.ts (hunk 1 above)
Comment (bfitzpat) [reply src/test/suite/completion.java.test.ts]: Reverted change

**Comment by bfitzpat** [comment]
So a couple of things came to light after working with @apupier for a bit...

1. The OUT folder had some older cached versions of copies of things which was causing the sinon.spy tests to fail.
2. The kubectlwatcher tests fail when you have a minikube or minishift instance running (deleting it makes those tests pass). 3. We should fix those tests with a mock to ensure that we get that resolved.

Thanks Aurelien.

**Comment by bfitzpat** [comment]
Now that we know the tests are fine and this was just a local error with the OUT folder, we can close this PR

**Pull request closed without merging**